packages/ra-core/src/routing/useGetPathForRecord.ts

```diff
@@ -0,0 +1,134 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useResourceContext, useResourceDefinition } from '../core';
+import { useCreatePath } from './useCreatePath';
+import { useRecordContext } from '../controller';
+import type { RaRecord } from '../types';
+import type { LinkToType } from './types';
+
+/**
+ * Get a path for a record, based on the current resource and the link type.
+ *
+ * Accepted link types are 'edit', 'show', a route string, false, or a function returning one of these types.
+ *
+ * @example
+ * // basic usage (leverages RecordContext, ResourceContext and ResourceDefinitionContext)
+ * const EditLink = () => {
+ *   const path = useGetRouteForRecord();
+ *   return path ? <Link to={path}>Edit</Link> : null;
+ * };
+ *
+ * // controlled mode
+ * const EditLink = ({ record, resource }) => {
+ *    const path = useGetRouteForRecord({ record, resource, link: 'edit' });
+ *    return path ? <Link to={path}>Edit</Link> : null;
+ * };
+ *
+ * // the link option can be a function
+ * const EditLink = ({ record, resource }) => {
+ *   const path = useGetRouteForRecord({ record, resource, link: (record, resource) => record.canEdit ? 'edit' : false });
+ *   return path ? <Link to={path}>Edit</Link> : null;
+ * };
+ *
+ * // the link option can be a function returning a promise
+ * const EditLink = ({ record, resource }) => {
+ *   const path = useGetRouteForRecord({ record, resource, link: async (record, resource) => {
+ *     const canEdit = await canEditRecord(record, resource);
+ *     return canEdit ? 'edit' : false;
+ *   }});
+ *   return path ? <Link to={path}>Edit</Link> : null;
+ * };
+ */
+export const useGetPathForRecord = <RecordType extends RaRecord = RaRecord>(
+    options: UseGetRouteForRecordOptions<RecordType>
+): string | false | undefined => {
+    const { link } = options || {};
+    const record = useRecordContext(options);
+    const resource = useResourceContext(options);
+    if (!resource) {
+        throw new Error(
+            'Cannot generate a link for a record without a resource. You must use useGetRouteForRecord within a ResourceContextProvider, or pass a resource prop.'
+        );
+    }
+    const createPath = useCreatePath();
+    const resourceDefinition = useResourceDefinition({ resource });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const linkFunc = useCallback(
```

Comment:
We cannot use `useEvent` here because we need to call the function right away, and `useEvent` throws an Error in that case.

Reply:
That means that if the function did change, it will never be updated here though right?

Reply:
yes indeed, that's a limitation I'm willing to accept